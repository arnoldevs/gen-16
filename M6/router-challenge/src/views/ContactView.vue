<template>
	<div class="contact">
		<h1 class="m-5">Ésta es una página de contacto</h1>
		<!-- Botón para volver al inicio -->
		<button class="btn btn-outline-secondary">
			<!-- Redirección a la página de inicio a través de RouterLink -->
			<RouterLink to="/" class="text-decoration-none text-dark"
				>Volver al inicio</RouterLink
			>
		</button>
	</div>
</template>

<style scoped>
@media (min-width: 1024px) {
	.contact {
		min-height: 100vh;
		display: flex;
		align-items: center;
		justify-content: center;
		flex-direction: column;
	}
}
</style>
