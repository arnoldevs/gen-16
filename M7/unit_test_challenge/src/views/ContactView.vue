<template>
	<div class="contact">
		<form action="">
			<input type="text" placeholder="Correo electrÃ³nico" />
			<textarea placeholder="Mensaje" rows="5" cols="30"></textarea>
			<button type="submit">Enviar</button>
		</form>
	</div>
</template>

<script>
export default {
	name: "ContactView",
};
</script>
<style scoped>
form {
	display: flex;
	flex-direction: column;
	align-items: center;
	justify-content: center;
	width: 100%;
	max-width: 400px;
	margin: 0 auto;
}
input,
textarea {
	padding: 10px;
	margin: 10px 0;
	width: 100%;
	border: 1px solid #ccc;
	border-radius: 5px;
}
button {
	background-color: #4caf50;
	color: white;
	padding: 10px 20px;
	border: none;
	border-radius: 5px;
	cursor: pointer;
	margin: 10px 0;
}
</style>
