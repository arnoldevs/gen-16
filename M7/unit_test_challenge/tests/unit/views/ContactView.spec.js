import { mount } from "@vue/test-utils";
import { createRouter, createWebHistory } from "vue-router";
import ContactView from "@/views/ContactView.vue";

describe("ContactView", () => {
	it("renders a contact form", async () => {
		const router = createRouter({
			history: createWebHistory(),
			routes: [
				{
					path: "/contact",
					name: "contact",
					component: ContactView,
				},
			],
		});
		router.push("/contact");
		await router.isReady();

		const wrapper = mount(ContactView, {
			global: {
				plugins: [router],
			},
		});

		// AsegÃºrate de que el formulario exista y tenga los elementos esperados
		expect(wrapper.find("form").exists()).toBe(true);
		expect(wrapper.find('input[type="text"]').exists()).toBe(true);
		expect(wrapper.find("textarea").exists()).toBe(true);
		expect(wrapper.find('button[type="submit"]').exists()).toBe(true);
	});

	it("matches snapshot", () => {
		const wrapper = mount(ContactView);
		expect(wrapper.html()).toMatchSnapshot();
	});
});
