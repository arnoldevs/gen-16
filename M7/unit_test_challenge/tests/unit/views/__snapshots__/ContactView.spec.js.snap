// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactView matches snapshot 1`] = `
<div class="contact">
  <form action=""><input type="text" placeholder="Correo electrÃ³nico"><textarea placeholder="Mensaje" rows="5" cols="30"></textarea><button type="submit">Enviar</button></form>
</div>
`;
